lucene/core/src/java/org/apache/lucene/index/MergePolicy.java

```diff
@@ -362,6 +366,30 @@ public void checkAborted() throws MergeAbortedException {
     public OneMergeProgress getMergeProgress() {
       return mergeProgress;
     }
+
+    /**
+     * Waits for this merge to be completed
+     * @return true if the merge finished within the specified timeout
+     */
+    boolean await(long timeout, TimeUnit timeUnit) {
+      try {
+        completable.get(timeout, timeUnit);
+        return true;
+      } catch (InterruptedException e) {
+        Thread.interrupted();
+        return false;
+      } catch (ExecutionException | TimeoutException e) {
+        return false;
+      }
+    }
+
+    boolean isDone() {
+      return completable.isDone();
+    }
+
+    boolean isCommitted() {
```

Comment:
Maybe rename to `isCompleted`?  `committed` is overloaded term -- could try to mean its files got `fsync'`d :)